### ISSUE TITLE
CLIP Learning Transferable Visual Models From Natural Language Supervision

### ISSUE THREAD

**Issue by Songinpyo** [Songinpyo/Paper-Record#13]
## ❓ Why → 해당 논문의 필요성, abstract
NLP분야에서 직접 annotation 하지 않고, 크롤링한 대량의 raw text 를 이용한 pretrain 기법이 큰 성공을 거두었다. (BERT)
이를 image분야에 적용하고자 한 것이 Natural Language Supervision 이다. (ConVIRT, VirTex)
 - Annotation에 필요한 노동력을 줄일 수 있다. 즉, 데이터셋을 쉽게 키울 수 있다.
 - Image representation만을 학습하는 것이 아닌 connection representations to language또한 학습하기 때문에 flexible한 zero-shot transfer learning이 가능해진다.
 
ConVIRT, VirTex는 쉽게 데이터셋을 키울 수 있는 장점을 활용하지 못하고 200,000장 내외의 크기 데이터셋을 사용하였다.
하지만, CLIP은 약 400M 개의 데이터를 이용하여 모델을 학습하였다.
**CLIP은 Simplied ConVIRT**를 이러한 데이터셋에서 학습시킨 것을 의미한다.
CLIP : Contrastive Language Image Pre training

## 💡 Key insights → 해당 논문의 핵심 아이디어
![image](https://user-images.githubusercontent.com/104220612/185908214-237880f9-0cb9-4d9d-99b6-61259e8abe7f.png)

Image와 Text를 각각 encoder에 넣어 embbeding을 뽑아낸다. 하나의 임베딩이 하나의 이미지와 텍스트의 임베딩을 의미한다.
상관관계를 볼때 대각선으로 되어있는 것만 positive 나머지를 negative로 보아서 consine similarity가 positive는 최대가 되도록, negative는 최소가 되도록 한다.
이러한 cosine similarity에 cross entropy를 적용하여 optimize한다. 따라서 image encoder와 text encoder를 동시에 학습시킨다.

📌 **Inference (Zero-shot prediction)**
![image](https://user-images.githubusercontent.com/104220612/185909028-e21c2ebe-3657-41e4-ab6c-cb4f2b0531db.png)
Downstream task에서 prediction할 때 pine tuning을 거치지 않고 사용한다.
CIFAR-10을 기준으로 설명하면 Text에는 classes가, image에는 prediction할 이미지를 넣고, 각각의 Text embbeding과 image의 embbeding의 cosine similarity score를 기준으로 prediction한다.
이 때, text는 단순히 'dog', 'cat' 등이 아니라 'A photo of a {label}' 형식으로 넣게 된다.

📌 **Prompt engineering**
일반적으로 두 가지의 문제를 잡기 위해서 이와 같은 text 변형을 거친다.
1. Polysemy 다의성, ( Crane : 크레인, 학 )
2. 학습과정의 text는 대부분 문장형이다.

단순히 text를 'A photo of a {label}' 형식으로 넣는 것 만으로 ImageNet의 acc가 1.3% 향상되었다.

📌 **Zero shot vs Few shot**
![image](https://user-images.githubusercontent.com/104220612/185910101-e09ba465-4e4a-4cb7-9ec9-f586e78d207e.png)
적어도 4개 이상의 데이터를 이용하여 pine tuning을 하였을 때 zero shot 보다 좋은 성능을 보였다.
충분한 학습이 이루어지지 않은 few shot은 역효과를 부를 수 있다.

## 🔗 Reference URL → 논문 링크
[Learning Transferable Visual Models From Natural Language Supervision](https://arxiv.org/pdf/2103.00020.pdf)
[유튜브 논문익는마을 김보민 논문 리뷰](https://www.youtube.com/watch?v=HkkaKI6NN-8)
